File: Examples/test-suite/python/cpp11_decltype_runme.py
import cpp11_decltype

a = cpp11_decltype.A()
a.i = 5
if a.i != 5:
    raise RuntimeError("Assignment to a.i failed.")

a.j = 10
if a.j != 10:
    raise RuntimeError("Assignment to a.j failed.")

n = a.get_number(5)
if n != 10:
    raise RuntimeError("get_number(5) should return 10.")

n = a.get_number(6)
if n != 0:
    raise RuntimeError("get_number(6) should return 0.")

b = cpp11_decltype.B()

if b.a != False:
    raise RuntimeError("b.a should be False")

if b.b != True:
    raise RuntimeError("b.b should be True")

if b.negate(True) != False:
    raise RuntimeError("b.negate(True) should return False")

if b.negate(False) != True:
    raise RuntimeError("b.negate(False) should return True")

File: Examples/test-suite/lua/cpp11_decltype_runme.lua
require("cpp11_decltype")

local a = cpp11_decltype.A()
a.i = 5
assert(a.i == 5, "Assignment to a:i failed.")
a.j = 10
assert(a.j == 10, "Assignment to a:j failed.")
local n = a:get_number(5)
assert(n == 10, "get_number(5) should return 10.")
n = a:get_number(6)
assert(n == 0, "get_number(6) should return 0.")

local b = cpp11_decltype.B()
assert(not b.a, "b.a should be false")
assert(b.b, "b.b should be true")
assert(not b:negate(true), "b.negate(true) should return false")
assert(b:negate(false), "b.negate(false) should return true")
